### PR TITLE
update nxos_nxapi module with minor enhancements

### PR DESCRIPTION
**Pull request by privateip**
- added new config argument
- added states present and absent
- update to use network shared modules

Tested on NXOS 7.3(0)D1(1)

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_nxapi.py [hunk 1]
```diff
@@ -31,238 +31,259 @@
...
+    required: false
+    default: true
+    choices:
+      - true
+      - false
+    aliases:
+      - enable_sandbox
+  config:
+    description:
+      - Specifies the device running-config to be used
```
Comment (gundalow) [inline network/nxos/nxos_nxapi.py]: Missing full stop

File: network/nxos/nxos_nxapi.py [hunk 2]
```diff
@@ -31,238 +31,259 @@
...
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: admin
+    password: admin
+
+- name: Enable NXAPI access with default configuration
+  nxos_nxapi:
+    provider: {{ cli }}
+
+- name: Enable NXAPI with no HTTP, HTTPS at port 9443 and sandbox disabled
```
Comment (gundalow) [inline network/nxos/nxos_nxapi.py:117]: From reading `set_https` it and like you do need to set 'enable_https: true'. Setting a port doesn't audio enable it. 

Http is `default=True`
Https is `default=False`

**Comment by gundalow** [comment]
needs_revision

**Comment by gundalow** [comment]
I also think the `state` parameter and options are non obviously named. Is this how NXAPI refer to it in there documentation?

**Comment by gregdek** [comment]
Thanks @privateip for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by privateip** [comment]
Updated the doc strings in the module to hopefully provide a little more clarity.  The module supports a number of different operating states as allowed by NXOS.

> I also think the state parameter and options are non obviously named. Is this how NXAPI refer to it in there documentation?

The state argument controls whether or not the configuration is present in the running config
- When the state=absent, the nxapi configuration is not present in the running config
- When the state=present, the nxapi configure is present in the running config

Independently, the NXAPI transport protocol states can be configured
- When http=yes, the http transport actively listens for connections on the configured http_port
- When http=no, the http transport is disabled
- When https=yes, the https transport actively listens for connections on the configured https_port
- When https=no, the https transport is disabled

ready_for_review

**Comment by gregdek** [comment]
Thanks @privateip. To the current maintainers, @privateip, @GGabriele, @gundalow please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Thanks for the documentation/examples update.

shipit

**Comment by gregdek** [comment]
Thanks again to @privateip for this PR, and thanks @privateip, @GGabriele, @gundalow for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Merged!

**Pull request merged**